README.md

```diff
@@ -1,11 +1,78 @@
 RateLimit.js
 ============
 [![Build
-Status](https://travis-ci.org/dudleycarr/ratelimit.js.svg)](https://travis-ci.org/dudleycarr/ratelimit.js)
+Status](https://travis-ci.org/dudleycarr/ratelimit.js.svg)](https://travis-ci.org/dudleycarr/ratelimit.js) [![npm version](https://badge.fury.io/js/ratelimit.js.svg)](http://badge.fury.io/js/ratelimit.js)
 
-A NodeJS library for rate limiting using sliding windows stored in Redis.
+A NodeJS library for efficiently rate limiting using sliding windows stored in Redis.
 
-### Express Middleware
+Features
+--------
+* Uses a sliding window for a rate limit rule
+* Multiple rules per instance
+* Multiple instances of RateLimit side-by-side for different categories of users.
+* Includes Express middleware
+
+Background
+----------
+See this excellent articles on how the sliding window rate limiting with Redis
+works:
+
+* [Introduction to Rate Limiting with Redis Part 1](http://www.dr-josiah.com/2014/11/introduction-to-rate-limiting-with.html)
+* [Introduction to Rate Limiting with Redis Part 2](http://www.dr-josiah.com/2014/11/introduction-to-rate-limiting-with_26.html)
+
+Install
+-------
+
+```
+npm install ratelimit.js
+```
+
+Usage
+-----
+
+Basic example:
+
+```javascript
+var RateLimit = require('ratelimt.js').RateLimit;
+var redis = require('redis');
+
+var client = redis.createClient();
+
+var rules = [
+  {interval: 1, limit: 5},
+  {interval: 3600, limit: 1000}
+  ];
+var limiter = RateLimit(client, rules);
+
+var showRatedLimited = function(err, isRateLimited) {
+  return console.log("Error: " + err);
+  console.log("Is rate limited? " + isRateLimited);
+};
+
+// Exceed rate limit.
+for(var i = 0; i < 10; i++) {
+  limiter.incr '127.0.0.1', function() {};
+}
+ratelimit.incr('127.0.0.1', isRateLimited);
+```
+
+
+Output:
+```
+Is rate limited? false
```

Comment:
This doesn't seem like the output for the code sample above since you pass an anonymous function to the first 10 calls of `limiter.incr`. The last line `ratelimit.incr` won't work either since `ratelimit` is undefined.